tests/unit/s2n_ecdsa_test.c

```diff
@@ -0,0 +1,175 @@
+    "ewQw//////////////////////////////////////////7/////AAAAAAAAAAD/\n"
+    "///8BDCzMS+n4j7n5JiOBWvj+C0ZGB2cbv6BQRIDFAiPUBOHWsZWOY2KLtGdKoXI\n"
+    "7dPsKu8DFQCjNZJqoxmieh0AiWpnc6SCes2scwRhBKqHyiK+iwU3jrHHHvMgrXRu\n"
+    "HTtii6ebmFn3QeCCVCo4VQLyXb9VKWw6VF44cnYKtzYX3kqWJixvXZ6Yv5KS3Cn4\n"
+    "9B29KJoUfOnaMRO18LjACmCxzh1+gZ16Qx18kOoOXwIxAP//////////////////\n"
+    "/////////////8djTYH0Ny3fWBoNskiwp3rs7BlqzMUpcwIBAQNiAATuRnqVT+re\n"
+    "o/6EEEW/pwLNDa7GrOZsTIRchHqjwrDrlnjtT7IcuWy5ALxEFGP0K0Xfh5kuBf4G\n"
+    "ebxhSZ690eYEVapsi8QDtvhu7V7jSgv8QQQRb4advo9CpsUWNFDHHGKjgb4wgbsw\n"
+    "HQYDVR0OBBYEFCTKHbjWPdLJjMlv3v5fH/vBTyWtMIGLBgNVHSMEgYMwgYCAFCTK\n"
+    "HbjWPdLJjMlv3v5fH/vBTyWtoV2kWzBZMQswCQYDVQQGEwJBVTETMBEGA1UECBMK\n"
+    "U29tZS1TdGF0ZTEhMB8GA1UEChMYSW50ZXJuZXQgV2lkZ2l0cyBQdHkgTHRkMRIw\n"
+    "EAYDVQQDEwlsb2NhbGhvc3SCCQCkJUe4OskXNjAMBgNVHRMEBTADAQH/MAkGByqG\n"
+    "SM49BAEDaQAwZgIxAJG5Ql52qUo+x9w7bJ3EZWpgt4jdblRkBvzpB1uX780h/6Wh\n"
+    "suQU36012pTQ6wTiNQIxAMJWqVLBmOJv3DSe4jsCeWwvWhWeItOy0fZ8fxW9w8VE\n"
+    "sXf3HEogY/fhiv5EUr+lcg==\n"
+    "-----END CERTIFICATE-----\n";
+
+static uint8_t private_key[] =
+    "-----BEGIN EC PRIVATE KEY-----\n"
+    "MIIB+gIBAQQwSJMrJhDMDa3hihWBYV/yyG7FLGPFxDhi+D6ZK6cXh/lzGdLztWmR\n" 
+    "UInVQV7mkO1RoIIBWzCCAVcCAQEwPAYHKoZIzj0BAQIxAP//////////////////\n"
+    "///////////////////////+/////wAAAAAAAAAA/////zB7BDD/////////////\n"
+    "/////////////////////////////v////8AAAAAAAAAAP////wEMLMxL6fiPufk\n"
+    "mI4Fa+P4LRkYHZxu/oFBEgMUCI9QE4daxlY5jYou0Z0qhcjt0+wq7wMVAKM1kmqj\n"
+    "GaJ6HQCJamdzpIJ6zaxzBGEEqofKIr6LBTeOscce8yCtdG4dO2KLp5uYWfdB4IJU\n"
+    "KjhVAvJdv1UpbDpUXjhydgq3NhfeSpYmLG9dnpi/kpLcKfj0Hb0omhR86doxE7Xw\n"
+    "uMAKYLHOHX6BnXpDHXyQ6g5fAjEA////////////////////////////////x2NN\n"
+    "gfQ3Ld9YGg2ySLCneuzsGWrMxSlzAgEBoWQDYgAE7kZ6lU/q3qP+hBBFv6cCzQ2u\n"
+    "xqzmbEyEXIR6o8Kw65Z47U+yHLlsuQC8RBRj9CtF34eZLgX+Bnm8YUmevdHmBFWq\n"
+    "bIvEA7b4bu1e40oL/EEEEW+Gnb6PQqbFFjRQxxxi\n"
+    "-----END EC PRIVATE KEY-----\n";
+
+static uint8_t unmatched_private_key[] =
+    "-----BEGIN EC PRIVATE KEY-----\n"
+    "MIIB+gIBAQQwuenHFMJsDm5tCQgthH8kGXQ1dHkKACmHH3ZqIGteoghhGow6vGmr\n" 
+    "xzA8gAdD2bJ0oIIBWzCCAVcCAQEwPAYHKoZIzj0BAQIxAP//////////////////\n"
+    "///////////////////////+/////wAAAAAAAAAA/////zB7BDD/////////////\n"
+    "/////////////////////////////v////8AAAAAAAAAAP////wEMLMxL6fiPufk\n"
+    "mI4Fa+P4LRkYHZxu/oFBEgMUCI9QE4daxlY5jYou0Z0qhcjt0+wq7wMVAKM1kmqj\n"
+    "GaJ6HQCJamdzpIJ6zaxzBGEEqofKIr6LBTeOscce8yCtdG4dO2KLp5uYWfdB4IJU\n"
+    "KjhVAvJdv1UpbDpUXjhydgq3NhfeSpYmLG9dnpi/kpLcKfj0Hb0omhR86doxE7Xw\n"
+    "uMAKYLHOHX6BnXpDHXyQ6g5fAjEA////////////////////////////////x2NN\n"
+    "gfQ3Ld9YGg2ySLCneuzsGWrMxSlzAgEBoWQDYgAE8oYPSRINnKlr5ZBHWacYEq4Y\n"
+    "j18l5f9yoMhBhpl7qvzf7uNFQ1SHzgHu0/v662d8Z0Pc0ujIms3/9uYxXVUY73vm\n"
+    "iwVevOxBJ1GL0usqhWNqOKoNp048H4rCmfyMN97E\n"
+    "-----END EC PRIVATE KEY-----\n";
+
+int main(int argc, char **argv)
```

Comment:
Good positive and negative tests. Do you think it's worth extending this to iterate across digest sizes: `0 -> MAX_DIGEST_LENGTH` ?